src/Serialization/BigInt.js

```diff
@@ -7,10 +7,12 @@ if (typeof BROWSER_RUNTIME != "undefined" && BROWSER_RUNTIME) {
   lib = require("@emurgo/cardano-serialization-lib-nodejs");
 }
 
-exports._BigInt_from_str = helper => str => {
-  try {
-    return helper.just(lib.BigInt.from_str(str));
-  } catch (_) {
-    return helper.nothing;
+exports._BigInt_from_str = maybe => str => {
+  // this is needed because try/catch overuse breaks runtime badly
+  // https://github.com/Plutonomicon/cardano-transaction-lib/issues/875
+  if (/^-?[0-9]+$/.test(str)) {
```

Comment:
I think we can try constructing JS BigInt to check CSL's BigInt string for validity instead of relying on a regexp.